### DISCUSSION TITLE
How to add custom snippets to LuaSnip when using LazyVim?

### DISCUSSION THREAD

**Discussion by martipares** [LazyVim/LazyVim#2608] [Q&A] [answered]
For the last week I was trying to setup LazyVim for LaTeX writing, but I have no idea of where to put my *. lua whit the custom LaTeX snippets.
Ideally, I would like a folder for different files whit the LaTeX snippets.
I would greatly appreciate the help.

**Comment by martipares** [comment]
Adding
config = function()
     require("luasnip.loaders.from_lua").load({paths = "~/.config/nvim/LuaSnip/"})
   end,
to the LuaSnip stander config (https://www.lazyvim.org/plugins/coding) in

{
  "L3MON4D3/LuaSnip",
<===
  build = (not jit.os:find("Windows"))
      and "echo 'NOTE: jsregexp is optional, so not a big deal if it fails to build'; make install_jsregexp"
    or nil,
  dependencies = {
.
.
.
seems to solve my problem.

**Comment by carb1d3** [comment]
can also remove everything after 
```
build = (not jit.os:find("Windows"))
and "echo 'NOTE: jsregexp is optional, so not a big deal if it fails to build'; make install_jsregexp"
```

**Reply by martipares** [reply]
Thank you, that is absolutely correct, the previous configuration was overdone and the snippets multiple times load, I will update the solution to something more minimalist that works better.

**Comment by martipares** [comment]
```lua
return {
  "L3MON4D3/LuaSnip",
  --  lazy = false,
  config = function()
    require("luasnip.loaders.from_lua").load({ paths = "~/.config/nvim/LuaSnip/" })
    require("luasnip").config.setup({ update_events = "TextChanged,TextChangedI" })
    require("luasnip").config.setup({ enable_autosnippets = true })
  end,
}
```

**Comment by martipares** [comment, marked as answer]
Little correction, unrelated to the main issue.

```lua
return {
  "L3MON4D3/LuaSnip",
--   lazy = false,
  config = function()
    require("luasnip.loaders.from_lua").load({paths = "~/.config/nvim/LuaSnip/"})
    require("luasnip").config.setup {
      update_events = 'TextChanged,TextChangedI',
      enable_autosnippets = true,
    }
  end,
}
```

**Reply by carb1d3** [reply]
Hey I lost my *.lua snippet file, can you please send your so I can build my cpp snippets from there.
Thanks

**Reply by martipares** [reply]
https://github.com/martipares/LuaSnip_01
I am not sure if this is what you wanted but in this repository I put my snippets for writing LaTeX and the prototype_01.lua file is lake a template (the function is for LaTeX specific things) so maybe it can be used.
